event-schema/event_schema.go

```diff
@@ -561,8 +583,10 @@ func getSchema(flattenedEvent map[string]interface{}) map[string]string {
 		if reflectType != nil {
 			schema[k] = reflectType.String()
 		} else {
-			schema[k] = "unknown"
-			pkgLogger.Errorf("[EventSchemas] Got invalid reflectType %+v", v)
+			if !(v == nil && !shouldCaptureNilAsUnknowns) {
+				schema[k] = "unknown"
+				pkgLogger.Errorf("[EventSchemas] Got invalid reflectType %+v", v)
```

Comment:
We are handling the nil type here. So, should we use debug instead of error?

Reply:
No. If its not nil we should ideally not get unknown. So if we got unknown, we should log it.